extension-style/src/main/java/com/mapbox/maps/extension/style/layers/LayerExt.kt

```diff
@@ -40,14 +40,18 @@ fun StyleManagerInterface.getLayer(layerId: String): Layer? {
 }
 
 /**
- * Tries to cast the Layer to T, throws ClassCastException if it's another type.
+ * Tries to cast the Layer to T.
  *
  * @param layerId the layer id
- * @return T
+ * @return T if layer is T, otherwise null
  */
-fun <T : Layer> StyleManagerInterface.getLayerAs(layerId: String): T {
-  @Suppress("UNCHECKED_CAST")
-  return getLayer(layerId) as T
+inline fun <reified T : Layer> StyleManagerInterface.getLayerAs(layerId: String): T? {
+  val layer = getLayer(layerId)
+  if (layer !is T) {
+    Logger.e("StyleLayerPlugin", "Given layerId = $layerId is not requested type in Layer")
```

Comment:
nit: introduce TAG in companion object and use that as first parameter of Logger.e (I do think we had a convention of prefixing with mbgl_ but better to check the codebase).

Reply:
`Given layerId = $layerId is not requested type in Layer` the error statement is a bit confusing. 
The layer itself could also just be null

Reply:
Any suggestion for error message ? :D

Reply:
> nit: introduce TAG in companion object and use that as first parameter of Logger.e (I do think we had a convention of prefixing with mbgl_ but better to check the codebase).

actually, this is not a class, it's an extension function file.

Reply:
Could we do something as:

```
private const val TAG = "mgbl-LayerUtils"
```
at bottom or top of file?

LayerUtils naming comes from the java name of the LayerExt file.